kclvm/api/src/service/service_impl.rs

```diff
@@ -472,20 +473,22 @@ impl KclvmServiceImpl {
     ///
     /// let serv = KclvmServiceImpl::default();
     /// let result = serv.rename(&RenameArgs {
+    ///     package_root: "./src/testdata/rename".to_string(),
     ///     symbol_path: "a".to_string(),
-    ///     file_paths: vec!["./src/testdata/rename/main.k".to_string()],
+    ///     file_paths: vec!["main.k".to_string()],
     ///     new_name: "a2".to_string(),
     /// }).unwrap();
     /// assert_eq!(result.changed_files.len(), 1);
     /// ```
     pub fn rename(&self, args: &RenameArgs) -> anyhow::Result<RenameResult> {
+        let pkg_root = args.package_root.clone();
```

Comment:
For non essential `clone` calls, you can use cargo clipping to check the relevant code.